packages/url_launcher/url_launcher_ios/example/ios/RunnerTests/URLLauncherTests.swift

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Flutter
+import XCTest
+
+@testable import url_launcher_ios
+
+final class URLLauncherTests: XCTestCase {
+  private var plugin: FLTURLLauncherPlugin!
+  private var launcher: FakeLauncher!
+
+  override func setUp() {
+    launcher = FakeLauncher()
+    plugin = FLTURLLauncherPlugin(launcher: launcher)
+  }
+
+  func testCanLaunchSuccess() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("good://url", error: &error)
+
+    XCTAssertTrue(result!.boolValue)
+    XCTAssertNil(error)
+  }
+
+  func testCanLaunchFailure() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("bad://url", error: &error)
+
+    XCTAssertFalse(result!.boolValue)
+  }
+
+  func testCanLaunchFailureWithInvalidURL() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("urls can't have spaces", error: &error)
+
+    XCTAssertNil(result)
+    XCTAssertNotNil(error)
```

Comment:
Ah I missed this. @chrisdlangham it's better to avoid force wrapping, since it would crash the whole test suite immediately. 

Using `?` allows XCTest to simply record the failure, continue with other tests, and give you a final complete report with all test failures. 

```
XCTAssertNotNil(error)
XCTAssertEqual(error?.code, "argument_error") // Use `?` rather than `!`
```

Reply:
i'll fix this.